packages/xo-server/src/api/vdi.js

```diff
@@ -119,6 +119,23 @@ set.resolve = {
 export async function migrate({ vdi, sr }) {
   const xapi = this.getXapi(vdi)
 
+  if (this.user.permission !== 'admin') {
+    const resourceSetId = reduce(
```

Comment:
This approach doesn't stop when we found the `resourceSet` which leads to a useless loop of all the collection.

If we support that a `VDI` can be attached to multiple `VMs` you can do a `for` + `break` otherwise simply access to the `VM` of the first `VBD`.